### PR TITLE
Add appearance for view, cell etc.

### PR DESCRIPTION
**Pull request by iON1k**
[Example](https://gist.github.com/iON1k/79fd02daabbc24fdecb345bd56ba58e4)

### CODE REVIEW

File: Sources/Extensions/TableDirector/TableRow+AppearanceExtension.swift [hunk 1]
```diff
@@ -20,20 +20,27 @@
...
+        }
+
+        action.id = updateAppearanceActionId
+        on(action)
+
+        return self
     }
+
+    func set(appearance: CellType.Appearance) {
+        _ = with(appearance: appearance)
```
Comment (scoreyou) [inline Sources/Extensions/TableDirector/TableRow+AppearanceExtension.swift]: use @discardableResult instead?

File: Sources/Extensions/TableDirector/TableRow+SeparatorsExtensions.swift [hunk 2]
```diff
@@ -22,24 +22,34 @@
 
 import TableKit
 
-public extension TableRow where CellType.T: SeparatorCellViewModel {
+fileprivate let configureSeparatorActionId = "TableRowConfigureSeparatorActionId"
```
Comment (scoreyou) [inline Sources/Extensions/TableDirector/TableRow+SeparatorsExtensions.swift]: just private?

File: Sources/Classes/Views/EmptyCell/EmptyCell.swift [hunk 3]
```diff
@@ -25,14 +25,35 @@ import TableKit
...
+    }
+
+    public func configure(appearance: Appearance) {
+        contentView.backgroundColor = appearance.color
+    }
+
+    public func configure(with _: Void) { }
+
+    private func setup() {
         backgroundColor = .clear
```
Comment (petropavel13) [inline Sources/Classes/Views/EmptyCell/EmptyCell.swift]: can be removed

File: Sources/Classes/Views/EmptyCell/EmptyCell.swift [hunk 4]
```diff
@@ -25,14 +25,35 @@ import TableKit
...
-        contentView.backgroundColor = viewModel.color
         selectionStyle = .none
```
Comment (petropavel13) [inline Sources/Classes/Views/EmptyCell/EmptyCell.swift]: move to `configure(appearance:)`?

File: Sources/Extensions/TableDirector/TableRow+AppearanceExtension.swift (hunk 1 above)
Comment (petropavel13) [inline Sources/Extensions/TableDirector/TableRow+AppearanceExtension.swift]: swap with and set bodies to avoid this?

Comment (petropavel13) [reply Sources/Extensions/TableDirector/TableRow+AppearanceExtension.swift]:
```swift
    func with(appearance: CellType.Appearance) -> Self {
        set(appearance: appearance)

        return self
    }

    func set(appearance: CellType.Appearance) {
        removeAction(forActionId: updateAppearanceActionId)

        let action = TableRowAction<CellType>(.configure) { options in
            options.cell?.configure(appearance: appearance)
        }

        action.id = updateAppearanceActionId
        on(action)
    }
```

File: Sources/Extensions/TableDirector/TableRow+SeparatorsExtensions.swift [hunk 5]
```diff
@@ -22,24 +22,34 @@
...
+
+        action.id = configureSeparatorActionId
+        on(action)
+
         return self
     }
 
     func set(separatorType: CellSeparatorType) {
-        item.set(separatorType: separatorType)
+        _ = with(separatorType: separatorType)
```
Comment (petropavel13) [inline Sources/Extensions/TableDirector/TableRow+SeparatorsExtensions.swift]: same about discardable result

File: Sources/Protocols/AppearanceProtocol.swift [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+//  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+//  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+//  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+//  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+//  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+//  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+//  THE SOFTWARE.
+//
+
+public protocol AppearanceProtocol {
```
Comment (petropavel13) [inline Sources/Protocols/AppearanceProtocol.swift]: doc comment?

File: Sources/Classes/Views/SeparatorRowBox/SeparatorRowBox.swift [hunk 7]
```diff
@@ -25,17 +25,19 @@ import TableKit
 /// Class that used to configure separators when multiply cells presented in one section
 /// Holds TableRow<T> with any model inherited from BaseCellViewModel
 public final class SeparatorRowBox {
-    /// Row `item`, that typed as BaseCellViewModel
-    public let viewModel: SeparatorCellViewModel
+    private let setSeparatorHandler: (CellSeparatorType) -> Void
+
+    public func set(separatorType: CellSeparatorType) {
```
Comment (petropavel13) [inline Sources/Classes/Views/SeparatorRowBox/SeparatorRowBox.swift]: doc comment

File: CHANGELOG.md [hunk 8]
```diff
@@ -37,3 +37,10 @@
 - **Change**: Rename `AnyBaseTableRow` class to `SeparatorRowBox`
 - **Change**: Move `anyRow` property from `EmptyCellRow` to `TableRow` extension and rename it to `separatorRowBox`.
 - **Change**: Move `configure(extreme: middle:)` method from `TableDirector` extension to `Array` extension and rename it to `configureSeparators(extreme: middle:)`
+
+## 0.5.15
+
+- **Add**: `AppearanceProtocol` which ensures that specific type can apply appearance to itself
+- **Add**: `with(appearance:)`,  `set(appearance:)`  methods to TableRow extansion
```
Comment (petropavel13) [inline CHANGELOG.md]: typo

**Pull request merged**